src/miner.cpp

```diff
@@ -439,6 +465,9 @@ void BlockAssembler::addPackageTxs()
             continue;
         }
 
+        // This transaction will make it in; reset the failed counter.
+        nConsecutiveFailed = 0;
```

Comment:
This is incremented whenever we fail (not just for failures after we're above a certain block weight), so it needs to be reset when adding a new tx.